tools/dumppdf.py

```diff
@@ -51,15 +55,17 @@ def dumpxml(out, obj, codec=None):
         out.write('</list>')
         return
 
-    if isinstance(obj, ((str,), bytes)):
+    if isinstance(obj, (str, bytes)):
         out.write('<string size="%d">%s</string>' % (len(obj), escape(obj)))
         return
 
     if isinstance(obj, PDFStream):
         if codec == 'raw':
-            out.write(obj.get_rawdata())
+            # Likely bug: writing bytes to text I/O.
+            out.write(obj.get_rawdata())  # type: ignore [arg-type]
```

Comment:
Can you add a test case that shows this crashes? If it does, we can change this code such that it doesn't.

Reply:
Ok. Note to self:
```
$ tools/dumppdf.py -o out -a --raw-stream samples/simple1.pdf
Traceback (most recent call last):
  File "tools/dumppdf.py", line 388, in <module>
    main()
  File "tools/dumppdf.py", line 378, in main
    dumppdf(
  File "tools/dumppdf.py", line 259, in dumppdf
    dumpallobjs(outfp, doc, codec, show_fallback_xref)
  File "tools/dumppdf.py", line 133, in dumpallobjs
    dumpxml(out, obj, codec=codec)
  File "tools/dumppdf.py", line 65, in dumpxml
    out.write(obj.get_rawdata())  # type: ignore [arg-type]
TypeError: write() argument must be str, not bytes

$ tools/dumppdf.py -o out -a --binary-stream samples/simple1.pdf
Traceback (most recent call last):
  File "tools/dumppdf.py", line 388, in <module>
    main()
  File "tools/dumppdf.py", line 378, in main
    dumppdf(
  File "tools/dumppdf.py", line 259, in dumppdf
    dumpallobjs(outfp, doc, codec, show_fallback_xref)
  File "tools/dumppdf.py", line 133, in dumpallobjs
    dumpxml(out, obj, codec=codec)
  File "tools/dumppdf.py", line 68, in dumpxml
    out.write(obj.get_data())  # type: ignore [arg-type]
TypeError: write() argument must be str, not bytes
```

Reply:
I've added a test for this (which fails), but it's hard to tell what the expected output actually is here -- both paths were trying to write raw binary data into the middle of an XML file. We could reach around the file and write raw bytes there, but would that be useful? We could choose a codec like base64, but would that be useful? I'm tempted to mark this as a known issue and leave it for someone else to decide how to fix it :)

Reply:
Yep, indeed. Let's not make this bigger than it already is. 

All the comments and type ignores are a big improvement, explicitly showing were we can improve.